### PR TITLE
fix: Storage plugin bug on Windows 

### PR DESCRIPTION
**Pull request by melund**
### Description

This PR attempts to fix the storage bug plugin test failures on Windows. 
Fixes: #2473

@johanneskoester, I am not sure this is how you want to fix the bug, but it should give you an idea what the problem is. 

Currently, on Windows all files in rules etc. are converted to forward slashes to avoid the 'backslash' mesh. However, the new storage plugin uses 'pathlib' for some paths internally. Converting pathlib.Path objects to strings gives paths with 'backslash'es on Windows. So I have added code to handle those cases. 

I have also improved the CI test setup on windows.  The setup time for conda environment creation is down from 12 min to 2.5 min on Windows. Much more is possible, but I have split that work into a separate PR (#2489)

The PR also does a few other things for local testing on Windows:

> Turns ou that "Git for Windows" have reorganized its internal Unix tools so they are no longer on path (Things like "cp", "touch", "cat" etc.) I guess it works in the cloud because the GitHub actions VM comes prepackaged with all tools on path from MSYS package. 
>
> A few tests still fail for me locally but work in the cloud, but I haven't dug into that.
### QC

* [x] The PR contains a test case for the changes or the changes are already covered by an existing test case.
* [x] The documentation (`docs/`) is updated to reflect the changes or this is not necessary (e.g. if the change does neither modify the language nor the behavior or functionalities of Snakemake).

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,3 +1,5 @@
+* text=auto eol=lf
```
Comment (melund) [inline .gitattributes:1]: This ensures files are checked-out with linux file endings on Windows.That is usually not the recommendation for windows. But this is the best option given how many input-output files in the tests system that are manipulated by posix tools.

File: pytest.ini [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[pytest]
+python_files = tests/tests.py tests/test_expand.py tests/test_io.py tests/test_schema.py tests/test_linting.py tests/test_executor_test_suite.py
```
Comment (melund) [inline pytest.ini:2]: The test suite contains plenty of files which doesn't follow the standard pytest conventions. This tripped the vs-code test plugin which tried to inspect non-test files which it thought caused it to crash.

**Comment by melund** [comment]
@johanneskoester. Test pass and it is ready for review.

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Thanks a lot! Sorry, the review was done since last week, but I forgot to hit the submit button. One thing below.

File: snakemake/io.py [hunk 3]
```diff
@@ -733,10 +733,10 @@ def apply_wildcards(self, wildcards):
...
                     rule=self.rule,
                 )
-
-            file_with_wildcards_applied = IOFile(
-                str(storage_object.local_path()), rule=self.rule
-            )
+            local_path_str = str(storage_object.local_path())
+            if ON_WINDOWS:
+                local_path_str = local_path_str.replace("\\", "/")
+            file_with_wildcards_applied = IOFile(local_path_str, rule=self.rule)
```
[the remark this answers is not in the document]
Comment (melund) [reply snakemake/io.py]: @johanneskoester, and have done that and pushed a new version.

**Comment by johanneskoester** [comment]
Seems like touching directories on windows does not work anymore with the changes here.

**Comment by melund** [comment]
> Seems like touching directories on windows does not work anymore with the changes here.

I will try to roll back so the check doesn't happen inside IOFile. The test were able to pass before I did that.

File: snakemake/io.py (hunk 3 above)
Comment (johanneskoester) [inline snakemake/io.py:739]: The Path object has a method `as_posix()` which does exactly what you do here. I don't understand though why we need this at all. What is the problem with having a windows path returned here?

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Comment by johanneskoester** [comment]
I have now implemented a simpler version of this in #2519. Thanks a lot for this work here, which pointer me to the right places for a simplified fix!

**Pull request closed without merging**